### PR TITLE
Add remote validator support to checker

### LINKED ISSUE #300: Add option for remote_validator
We should add an option in the `csaf_downloader` and `csaf_checker` to run a remote validator.

### PR DESCRIPTION
**Pull request by s-l-teichmann**
This PR adds support for remote validation to the checker.

This partly resolves #300

### CODE REVIEW

File: cmd/csaf_checker/main.go [hunk 1]
```diff
@@ -39,6 +39,10 @@ type options struct {
 	Rate       *float64 `long:"rate" short:"r" description:"The average upper limit of https operations per second"`
 	Years      *uint    `long:"years" short:"y" description:"Number of years to look back from now" value-name:"YEARS"`
 
+	RemoteValidator        string   `long:"validator" description:"URL to validate documents remotely" value-name:"URL"`
+	RemoteValidatorCache   string   `long:"validatorcache" description:"FILE to cache remote validations" value-name:"FILE"`
+	RemoteValidatorPresets []string `long:"validatorpreset" description:"One or more presets to validate remotely"`
```
Comment (tschmidtb51) [inline cmd/csaf_checker/main.go:44]: Maybe document that the default preset is `mandatory`.

**Review by tschmidtb51 — DISMISSED** [review]
Beside the documentation comment.
LGTM and works as expected.

File: cmd/csaf_checker/main.go (hunk 1 above)
Comment (s-l-teichmann) [reply cmd/csaf_checker/main.go]: @tschmidtb51 2a40ab6393d1bf445c51255598366f73cd454f0c let `validatorpreset`  defaults to `mandatory`.
This is displayed with `-h/--help` and is therefore mentioned in the docs.  
I am going to do the same in the sister PR #312 to let the downloader behave the same way.

BTW: I have added `[OPTIONS] domain...' to the usage output to match the downloader.

**Review by tschmidtb51 — APPROVED** [review]
LGTM and works.

File: cmd/csaf_checker/main.go (hunk 1 above)
Comment (tschmidtb51) [reply cmd/csaf_checker/main.go]: Ok. I closed #321 in favor of this one here.

**Review by tschmidtb51 — APPROVED** [review]
LGTM

File: docs/csaf_checker.md [hunk 2]
```diff
@@ -15,8 +16,12 @@ Application Options:
   -v, --verbose                  Verbose output
   -r, --rate=                    The average upper limit of https operations per second
   -y, --years=YEARS              Number of years to look back from now
+      --validator=URL            URL to validate documents remotely
+      --validatorcache=FILE      FILE to cache remote validations
+      --validatorpreset=         One or more presets to validate remotely (default: mandatory)
   -H, --header=                  One or more extra HTTP header fields
```
Comment (tschmidtb51) [inline docs/csaf_checker.md:22]:
```suggestion

File: docs/csaf_checker.md [hunk 3]
```diff
@@ -15,8 +16,12 @@ Application Options:
   -v, --verbose                  Verbose output
   -r, --rate=                    The average upper limit of https operations per second
   -y, --years=YEARS              Number of years to look back from now
+      --validator=URL            URL to validate documents remotely
```
Comment (tschmidtb51) [inline docs/csaf_checker.md:19]:
```suggestion
  -H, --header=                  One or more extra HTTP header fields
      --validator=URL            URL to validate documents remotely

**Review by tschmidtb51 — COMMENTED** [review]
I accidentally switched the wrong way...

**Review by tschmidtb51 — APPROVED** [review]
LGTM

**Review by bernhardreiter — APPROVED** [review]
LGTM

**Pull request merged**